src/utilities/buildASTSchema.js

```diff
@@ -211,6 +216,16 @@ export function buildASTSchema(
     return def.interfaces.map(inter => produceTypeDef(inter));
   }
 
+  function makeInterfaceImplementations(inter: InterfaceDefinition) {
```

Comment:
This is `O(N^2)` since you need to filter down the AST and produceTypeDef for the implementing types for every interface you encounter, (so really `O(I*N)`)

Here's how I did it in https://github.com/graphql/graphql-js/blob/master/src/utilities/buildClientSchema.js#L302

On a side note, you may consider comparing this file to that file to see if there are any other gaps